File: me/seeking/value/Option.java
package me.seeking.value;



import java.util.function.Supplier;

public class Option extends Value<Boolean> {

    public Option(String name, Boolean enabled) {
        super(name, enabled, () -> true);
    }

    public Option(String name, Boolean enabled, Supplier<Boolean> visitable) {
        super(name, enabled, visitable);
    }

    @Override
    public Boolean getValue() {
        return this.isVisitable() && super.getValue();
    }
}

File: src/main/java/me/seeking/value/Option.kt
package me.seeking.value

import java.util.function.Supplier

/**
 * This file is a part o Seeking
 */

class Option(name: String?, enabled: Boolean?) : Value<Boolean?>(name!!, enabled, Supplier<Boolean> { true }) {
    fun getValue(): Boolean {
        return isVisitable() && super.value!!
    }
}

